### PR TITLE
refactor(core): Remove clickmod support from Angular.

### PR DESCRIPTION
**Pull request by iteriani**
This was an old feature that mapped shift + click (et al) to "clickmod". This doesn't really make sense to add to Angular, so let's remove it.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by thePunderWoman — APPROVED** [review]
LGTM

reviewed-for: fw-core, primitives, public-api

**Review by AndrewKushnir — APPROVED** [review]
Reviewed-for: public-api

File: packages/core/src/event_delegation_utils.ts [hunk 1]
```diff
@@ -115,6 +115,6 @@ export const initGlobalEventDelegation = (
   const eventContract = (eventContractDetails.instance = new EventContract(
     new EventContractContainer(document.body),
   ));
-  const dispatcher = new EventDispatcher(invokeRegisteredListeners);
+  const dispatcher = new EventDispatcher(invokeRegisteredListeners, /** clickModSupport */ false);
```
Comment (tbondwilkinson) [inline packages/core/src/event_delegation_utils.ts:118]: /* clickModSupport= */

**Review by tbondwilkinson — APPROVED** [review]

**Comment by iteriani** [comment]
TESTED=[TGP](https://fusion2.corp.google.com/presubmit/658063554/OCL:658063554:BASE:658281891:1722493800864:58b1f3f7;groups=PossiblyNewlyFailing/targets)

**Comment by thePunderWoman** [comment]
This PR was merged into the repository by commit 2a915d1912d6b3e9ba05b57a0fcd993a2f3f0221.

The changes were merged into the following branches: main, 18.1.x

**Pull request closed without merging**